File: frontend/types/tender.ts
export interface Tender {
  id: string;
  tender_id: string;
  source: string;
  title: string;
  description: string;
  authority: string;
  location: string;
  category: string;
  budget_min: number | null;
  budget_max: number | null;
  budget_raw: string;
  published_at: string | null;
  deadline: string | null;
  deadline_raw: string;
  status: string;
  source_url: string;
  ai_summary: string | null;
  ai_risk: string | null;
  ai_eligibility: string | null;
  scraped_at: string | null;
  created_at: string;
}

export interface TenderListResponse {
  items: Tender[];
  total: number;
  page: number;
  per_page: number;
  pages: number;
}

File: backend/app/schemas/tender.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class TenderResponse(BaseModel):
    id: str
    tender_id: str
    source: str
    title: str
    description: str
    authority: str
    location: str
    category: str
    budget_min: Optional[float]
    budget_max: Optional[float]
    budget_raw: str
    published_at: Optional[datetime]
    deadline: Optional[datetime]
    deadline_raw: str
    status: str
    source_url: str
    ai_summary: Optional[str]
    ai_risk: Optional[str]
    ai_eligibility: Optional[str]
    match_score: Optional[float] = None
    match_reasoning: Optional[str] = None
    scraped_at: Optional[datetime]
    created_at: datetime

    class Config:
        from_attributes = True


class TenderListResponse(BaseModel):
    items: list[TenderResponse]
    total: int
    page: int
    per_page: int
    pages: int